test/topology_experimental_raft/test_tombstone_gc.py

```diff
@@ -198,3 +198,39 @@ async def verify_tombstone_gc(tombstone_mark, timeout=30):
 
             # test #3: the tombstones are cleaned up after the node is started again
             await verify_tombstone_gc(tombstone_mark)
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail(reason="Issue #21117")
+async def test_group0_state_id_failure(manager: ManagerClient):
+    """
+    Issue #21117 regression test.
+
+    Make sure that the "endpoint_state_map does not contain endpoint" warning is not triggered when the state_id is updated.
+
+    Arrange:
+        - start 1 node
+    Act:
+        - restart the node
+    Assert:
+        - the "endpoint_state_map does not contain endpoint" didn't appear in the log
+    """
+
+    # Start the node
+
+    srv = await manager.server_add()
+
+    cql = manager.get_cql()
+    await wait_for_cql_and_get_hosts(cql, [srv], time.time() + 60)
+
+    # Restart the node
+
+    await manager.server_restart(srv.server_id)
+    await wait_for_cql_and_get_hosts(cql, [srv], time.time() + 60)
+
+    # Check the log
+
+    log = await manager.server_open_log(srv.server_id)
+    matches = await log.grep(r"Fail to apply application_state.*endpoint_state_map does not contain endpoint .* application_states = {GROUP0_STATE_ID")
```

Comment:
`application_states = {GROUP0_STATE_ID`

hmm, could multiple states be listed here? If so, there probably should be `.*` after `{`. If not, then this is OK